internal/service/cognitoidp/user_pool_domain.go

```diff
@@ -156,23 +167,29 @@ func resourceUserPoolDomainUpdate(ctx context.Context, d *schema.ResourceData, m
 	var diags diag.Diagnostics
 	conn := meta.(*conns.AWSClient).CognitoIDPClient(ctx)
 
-	input := &cognitoidentityprovider.UpdateUserPoolDomainInput{
-		CustomDomainConfig: &awstypes.CustomDomainConfigType{
-			CertificateArn: aws.String(d.Get(names.AttrCertificateARN).(string)),
-		},
+	timeout := 5 * time.Minute
+	input := cognitoidentityprovider.UpdateUserPoolDomainInput{
 		Domain:     aws.String(d.Id()),
 		UserPoolId: aws.String(d.Get(names.AttrUserPoolID).(string)),
 	}
 
-	_, err := conn.UpdateUserPoolDomain(ctx, input)
+	if d.HasChange(names.AttrCertificateARN) {
+		timeout = 60 * time.Minute
```

Comment:
Perhaps pull this comment back to from the deleted section below to preserve context on the variable timeout?

```suggestion
		timeout = 60 * time.Minute // Certificate ARN updates on custom domains take more time to become active.
```